### PR TITLE
feat: offchain bids

### PR DESCRIPTION
**Pull request by meelrossi**
Add offchain bids logic. This is the sum of a series of PRs
https://github.com/decentraland/marketplace/pull/2273
https://github.com/decentraland/marketplace/pull/2272
https://github.com/decentraland/marketplace/pull/2271
https://github.com/decentraland/marketplace/pull/2270
https://github.com/decentraland/marketplace/pull/2269
https://github.com/decentraland/marketplace/pull/2268

### CODE REVIEW

File: webapp/src/components/Bid/AcceptButton/AcceptButton.tsx [hunk 1]
```diff
@@ -21,12 +22,15 @@ const AcceptButton = (props: Props) => {
...
+  const isValidSeller = assetOwner && assetOwner === userAddress
+  const isItemAvailable = !!asset && (isNFT(asset) || asset.available > 0)
 
-  const isDisabled = isCurrentlyLocked || !nft || isLoadingFingerprint || hasInsufficientMANA || !isValidFingerprint || !isValidSeller
+  const isDisabled =
+    isCurrentlyLocked || !asset || isLoadingFingerprint || hasInsufficientMANA || !isValidFingerprint || !isValidSeller || !isItemAvailable
 
   let button = (
-    <Button primary disabled={isDisabled} onClick={onClick}>
+    <Button size="small" primary disabled={isDisabled} onClick={onClick}>
```
Comment (LautaroPetaccio) [inline webapp/src/components/Bid/AcceptButton/AcceptButton.tsx:33]: Should this be a `NetworkButton`?

File: webapp/src/components/AssetPage/AssetPage.tsx [hunk 2]
```diff
@@ -15,7 +15,7 @@ import { Props } from './AssetPage.types'
 import './AssetPage.css'
 
 const AssetPage = ({ type }: Props) => {
-  const renderItemDetail = useCallback((item: Item) => <ItemDetail item={item} />, [])
+  const renderItemDetail = useCallback((item: Item) => <ItemDetail item={item} isBidsOffchainEnabled={false} />, [])
```
Comment (LautaroPetaccio) [inline webapp/src/components/AssetPage/AssetPage.tsx:18]: Should this work with a FF?

File: webapp/src/components/AssetPage/SaleRentActionBox/SaleRentActionBox.tsx [hunk 3]
```diff
@@ -238,20 +240,7 @@ const SaleRentActionBox = ({
...
-                              [styles.bid_manage_in_builder]: isTenant && !rentalHasEnded
-                            })}
-                            disabled={isNFTPartOfAState}
-                            primary={!order && !(isTenant && !rentalHasEnded)}
-                            secondary={true}
-                            fluid
-                          >
-                            {t('asset_page.actions.bid')}
-                          </Button>
+                          <BidButton asset={nft} alreadyBid={userHasAlreadyBidsOnNft} disabled={isNFTPartOfAState} />
```
Comment (LautaroPetaccio) [inline webapp/src/components/AssetPage/SaleRentActionBox/SaleRentActionBox.tsx:243]: The button seems to change given this condition:
`!order && !(isTenant && !rentalHasEnded)`
Is this something we want to remove?

**Review by LautaroPetaccio — APPROVED** [review]

File: webapp/src/components/Bid/AcceptButton/AcceptButton.tsx (hunk 1 above)
Comment (meelrossi) [reply webapp/src/components/Bid/AcceptButton/AcceptButton.tsx:33]: There is no need to be in the polygon network as we can use metatransactions

File: webapp/src/components/AssetPage/SaleRentActionBox/SaleRentActionBox.tsx (hunk 3 above)
Comment (meelrossi) [reply webapp/src/components/AssetPage/SaleRentActionBox/SaleRentActionBox.tsx:243]: its only if its primary or secondary type button and we completely change the look and feel of the make offer button so we hsould ignore this

**Pull request merged**